tests/unit/features/partitions/test_parts.py

```diff
@@ -0,0 +1,206 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from textwrap import dedent
+
+import pytest
+
+from craft_parts import parts
+from craft_parts.parts import Part
+from tests.unit import test_parts
+
+
+class TestPartSpecs(test_parts.TestPartSpecs):
```

Comment:
This (and the other "no-op" classes) is testing that the existing behavior is still working if the partitions feature is enabled but the project doesn't use partitions, right? Can you add a comment near here saying that? To remind our future selves.

Reply:
Good idea, I added a comment.

You are correct about everything except for "the partitions feature is enabled but the project doesn't use partitions".  So to be precise (I italicized the small difference):

As you said these no-op classes verify you can enable the partitions feature and create Parts with no change in behavior.  Now, once your application gets to the point of initializing a LifecycleManager and you provide it the list of parts, then the application _must provide a list of partitions to the LifecycleManager when the feature is enabled_.